csrc/scheduler/matmul.cpp

```diff
@@ -473,10 +473,17 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
   TensorView* c = outputs[0]->as<TensorView>();
 
   // Collect mma swizzle info
-  const auto layout = mma_ops.front()->layout().value();
+  auto mma = mma_ops.front();
+
+  // Note: additional temp variable defined to satify clang-tidy requirements
```

Comment:
If you change the above check from
```c++
mma_ops.front()->layout().has_value(),
```
into
```c++
layout_opt.has_value()
```
Clang-tidy will be happy. Unfortunately, clang-tidy is not smart enough to infer that two calls of `container->front()` results in the same value...

Reply:
I tried a number of combinations but clang-dity will be happy only if there is an explicit and separate check of `std::optional` (it can be via `bool()` or `has_value()`). I hoped that this will work
```
const auto layout = mma->layout().has_value() ? mma->layout().value() : MmaOptions::MmaLayout::TN;
```
but not, it will translate into data prefetching before checking condition.
The working solution was to add `TORCH_INTERNAL_ASSERT` to do the implicit check, with `bool()` operator, it variable is initialized.

Is the current code not good enough?

Reply:
The only problem is you `TORCH_INTERNAL_ASSERT` checked `has_value()` twice, once in line 467, and another one in line 480. And this is redundant.

Reply:
oh, I missed that, sorry for the noise - I will have this updated and simplified in a moment.
Thanks!